### PR TITLE
feature / DAO-677

### PR DESCRIPTION
**Pull request by Rekard0**
- Aggregate balances
- Add timestamp to vault transfers & balances
- Update unit testing

### CODE REVIEW

File: packages/subgraph/schema.graphql [hunk 1]
```diff
@@ -10,9 +10,10 @@ type ERC20Token @entity {
 
 type ERC20Balance @entity {
```
Comment (nivida) [inline packages/subgraph/schema.graphql]: Cause we have ETH handled with the same struct can we also name it ``Balance``.. wdyt?

File: packages/subgraph/src/dao.ts [hunk 2]
```diff
@@ -28,22 +29,105 @@ export function handleExecuted(event: Executed): void {
   // TODO:
 }
 
+function updateERC20Balance(
```
Comment (nivida) [inline packages/subgraph/src/dao.ts]: Same naming question here :))

File: packages/subgraph/src/dao.ts [hunk 3]
```diff
@@ -28,22 +29,105 @@ export function handleExecuted(event: Executed): void {
...
+  let tokenContract = GovernanceWrappedERC20.bind(token);
+  let daoBalance = tokenContract.try_balanceOf(daoAddress);
+  if (!daoBalance.reverted) {
+    entity.balance = daoBalance.value;
+  }
+  entity.lastUpdated = timestamp;
+  entity.save();
+}
+
+function updateEthBalance(
```
Comment (nivida) [inline packages/subgraph/src/dao.ts]: I think we have here some code duplication we could remove :-)

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/subgraph/src/dao.ts (hunk 2 above)
Comment (Rekard0) [reply packages/subgraph/src/dao.ts]: amended

File: packages/subgraph/schema.graphql (hunk 1 above)
Comment (Rekard0) [reply packages/subgraph/schema.graphql]: yes, makes sense , amended

File: packages/subgraph/src/dao.ts (hunk 3 above)
Comment (Rekard0) [reply packages/subgraph/src/dao.ts]: yes, amended

**Review by nivida — APPROVED** [review]

**Pull request merged**